### PR TITLE
feat: Add onComplete callback to `AnimationWidget`

### PR DESCRIPTION
**Pull request by willhlas**
# Description

Adds optional onComplete callback to `SpriteAnimationWidget` for `asset` constructor.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/widgets/animation_widget.dart [hunk 1]
```diff
@@ -36,7 +39,8 @@ class SpriteAnimationWidget extends StatelessWidget {
   })  : _animationFuture = animation,
         _animationTicker = animationTicker,
         errorBuilder = null,
-        loadingBuilder = null;
+        loadingBuilder = null,
+        _onComplete = null;
```
Comment (spydon) [inline packages/flame/lib/src/widgets/animation_widget.dart:43]: Why not expose onComplete from the default constructor too?

File: packages/flame/lib/src/widgets/animation_widget.dart [hunk 2]
```diff
@@ -48,22 +52,27 @@ class SpriteAnimationWidget extends StatelessWidget {
     required String path,
     required SpriteAnimationData data,
     Images? images,
+    VoidCallback? onComplete,
```
Comment (spydon) [inline packages/flame/lib/src/widgets/animation_widget.dart:55]:
```suggestion
    this.onComplete,
```
I don't think onComplete needs to be private, and we can set it directly here.

**Review by spydon — COMMENTED** [review]
Just some small nits, the idea is good!

File: packages/flame/lib/src/widgets/animation_widget.dart (hunk 1 above)
Comment (erickzanardo) [reply packages/flame/lib/src/widgets/animation_widget.dart:43]: The default constructor receives a concrete instance of a `SpriteAnimation`, which can have the callback in it. So I believe we don't need that callback there?

But at the same time, I agree that there is no harm in having it there as well

**Review by ufrshubham — APPROVED** [review]
Not sure how useful the other callbacks will be in this widget, but it sure would look more complete if `onStart` and `onFrame` are exposed too. Anyways, current changes look fine to me 👍🏼

File: packages/flame/lib/src/widgets/animation_widget.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/widgets/animation_widget.dart:43]: True, I'm fine with either way then

**Comment by erickzanardo** [comment]
> Not sure how useful the other callbacks will be in this widget, but it sure would look more complete if `onStart` and `onFrame` are exposed too. Anyways, current changes look fine to me 👍🏼

Yeah, good call. I think we should expose either way, there may always have a use case that we can't see right away.

I would argue that we should do each callbacks in a separate PR, wdyt?

**Review by wolfenrain — APPROVED** [review]

File: packages/flame/lib/src/widgets/animation_widget.dart (hunk 1 above)
Comment (erickzanardo) [reply packages/flame/lib/src/widgets/animation_widget.dart:43]: I would keep as is then :D

**Review by st-pasha — APPROVED** [review]

File: packages/flame/lib/src/widgets/animation_widget.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/lib/src/widgets/animation_widget.dart:55]: @willhlas, did you see this?

Comment (willhlas) [reply packages/flame/lib/src/widgets/animation_widget.dart]: @spydon just pushed the update, sorry for the delay!

**Review by spydon — APPROVED** [review]

**Pull request merged**